### PR TITLE
Run go mod tidy in workflow

### PR DESCRIPTION
**Pull request by mukundansundar**
# Description

Run go mod tidy and check difference and fail if changes found in workflow.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by tcnghia — APPROVED** [review]
Looks good to me. Thanks for making this change!

In the future if we decide to use Dependabot it can automatically tidy for us.

**Comment by youngbupark** [comment]
@mukundansundar This is good. What happens if `go mod tidy` cleaned up the pkg incorrectly without our control ? the binary could be different from what we tested.. Instead, we can run `go mod tidy` only for PR build and if there is the change required, then you can return non zero exit code to let PR author notify that `go mod tidy` is required.

I trust `go mod tidy` behavior, but there is a chance that this command can change go.mod and go.sub out of our control. 

@yaron2 @tcnghia any thoughts?

**Comment by tcnghia** [comment]
> @mukundansundar This is good. What happens if `go mod tidy` cleaned up the pkg incorrectly without our control ? the binary could be different from what we tested.. Instead, we can run `go mod tidy` only for PR build and if there is the change required, then you can return non zero exit code to let PR author notify that `go mod tidy` is required.
> 
> I trust `go mod tidy` behavior, but there is a chance that this command can change go.mod and go.sub out of our control.
> 
> @yaron2 @tcnghia any thoughts?

From my reading of the workflow, this is exactly as you thought.

**Comment by youngbupark** [comment]
> > @mukundansundar This is good. What happens if `go mod tidy` cleaned up the pkg incorrectly without our control ? the binary could be different from what we tested.. Instead, we can run `go mod tidy` only for PR build and if there is the change required, then you can return non zero exit code to let PR author notify that `go mod tidy` is required.
> > I trust `go mod tidy` behavior, but there is a chance that this command can change go.mod and go.sub out of our control.
> > @yaron2 @tcnghia any thoughts?
> 
> From my reading of the workflow, this is exactly as you thought.

Oh right. I missed the change in makefile. Apologies..

**Review by youngbupark — DISMISSED** [review]
👍

File: .github/workflows/dapr.yml [hunk 1]
```diff
@@ -71,6 +71,9 @@ jobs:
         uses: golangci/golangci-lint-action@v2.2.1
         with:
           version: ${{ env.GOLANGLINT_VER }}
+      - name: Run go mod tidy check diff
+        if: matrix.target_arch == 'amd64' && matrix.target_os == 'linux'
+        run: make modtidy check-diff
```
Comment (tcnghia) [inline .github/workflows/dapr.yml:76]: If you want to output an error message that doesn't require looking at the log of failing workflow:

```
echo "::error file=optional.js,line=10,col=15::Something went wrong"
```
(all options are optional)
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-error-message

**Review by youngbupark — APPROVED** [review]

**Pull request merged**